### ISSUE TITLE
When the table's "SHARD KEY" is set, the max number of rows in the table is equal to "max_rows * shard_count". The actual number of rows in the table may be much larger than the upper limit set.

### ISSUE THREAD

**Issue by thzyyx** [heavyai/heavydb#297]
When the table's "SHARD KEY" is set, the max number of rows in the table is equal to "max_rows * shard_count". The actual number of rows in the table may be much larger than the upper limit set.This caused me to lose control of the number of rows in the table

**Comment by thzyyx** [comment]
I found this problem when analyzing the data elimination mechanism. The document tells me,"When this limit is reached, the oldest fragment is removed. Default = 2^62.".I set  to smaller fragment_size and max_rows of table. If you insert more data than "max_rows"  and to view the file on disk, it will be unchanged . I want the file size to be reduced. In the process, I printed out the frags of the table. If the table is not set to "shard," the number of frags（frag ID number） the table is equal to "max_rows/fragment_size." When "shard" is set,  each frag ID has "shard_count" actual frag. The elimination of the old frag is based on the actual frag, not the frag ID.This is equivalent to expanding the capacity of the table

**Comment by dwayneberry** [comment]
@thzyyx Yes you are correct that sharding will increase the number of rows the total table will contain

If I create a table with following WITH clause

WITH (SHARD_COUNT = 16, MAX_ROWS = 10000000)

Means that actual number of rows that will load into the db before old stuff gets dropped is actually potentially 160,000,000

Which is calculated as
```
max_rows * shard_count
```
As the max row is managed per table and the shards are each physical tables hidden behind a logical wrapper.

This is the behavior we currently have, not necessarily very intuitive

**Comment by psilvest** [comment]
This issue has been inactive for a while, and the project has changed substantially. If this still occurs on the latest release, please add reproduction steps and environment details.
